base/abstractarray.jl

```diff
@@ -392,7 +392,11 @@ julia> stride(A,3)
 12
 ```
 """
-stride(A::AbstractArray, k::Integer) = strides(A)[k]
+function stride(A::AbstractArray, k::Integer) 
+    k < 1 && return 1
+    k ≤ ndims(A) && return strides(A)[k] 
+    length(A)
```

Comment:
So now `stride(1:5, 0)` and `stride(1:5, 2)` will return values but `stride(1:5, 1)` will throw an error? To me it seems better to throw a `MethodError` in all cases.

Reply:
I've modified it so it always calls `strides` now, producing the errors consistently now.